Learn.md

```diff
@@ -0,0 +1,178 @@
+```
+5. **Push Your Changes** to your fork: Upload your committed changes to your GitHub fork.
+   ```bash
+   git push origin <branch_name>
+   ```
+6. **Create a Pull Request ✅**: Propose your changes to be merged into the original repository.
+   <br>
+   
+![image](https://github.com/debangi29/ConvoForm/assets/117537653/8803d568-0c43-4972-8a95-b9b91b9bf77e)
+
+
+### Development Workflow
+- Always work on a new branch for each issue or feature.
+- Keep your branch up to date with the main repository's master branch.
+- Write clear and descriptive commit messages.
+- Test your changes thoroughly before submitting a pull request.
+
+### Issue Report Process
+1. Go to the project's issues section.
+2. Select the appropriate template for your issue.
+3. Provide a detailed description of the issue.
+4. Wait for the issue to be assigned before starting to work on it.
+
+### **Pull Request Process 🚀**
+
+1. Ensure that you have self-reviewed your code.
+2. Make sure you have added the proper description for the functionality of the code.
+3. I have commented on my code, particularly in hard-to-understand areas.
+4. Add screenshots it helps in review.
+5. Submit your PR by giving the necesarry information in PR template and hang tight we will review it really soon.
+
+# Setting Up on your machine ⚙️
+
+Follow these steps to set up the project locally on your machine.
+
+### Prerequisites
+
+- Make sure you have the following installed:
+
+    - [Node.js](https://nodejs.org/) (v18.17.1 or higher)
+    - [pnpm](https://pnpm.io/) (v8.14.3 or higher)
+    - [Git](https://git-scm.com/)
+
+- You need a PostgreSQL database instance to store the customer data. Create a free superbase Postgres instance at [Supabase](https://supabase.com).
+
+- [Clerk](clerk.com) is used for Authentication and User Management. Please make sure you Enable organizations in Organizations Settings within Clerk settings.
+
+- Get your OpenAI key at [OpenAI Dashboard](https://platform.openai.com/api-keys)
+
+- You need a Redis Database from [Upstash](https://upstash.com) (Optional if you want rate limiting)
+
+- Application Performance Monitoring & Error Tracking is done using [Sentry](https://sentry.io)
+
+1. #### Clone the Repository
+
+```bash
+git clone https://github.com/growupanand/ConvoForm.git
+cd ConvoForm
+```
+
+2. #### Install Dependencies
+
+```bash
+pnpm install
+```
+
+3. #### Configuration environment
+
+    1. Copy the .env.example file to .env.local file.
+       ```bash
+       cp .env.example .env.local
+       ```
+    2. Open the `.env.local` file and update the necessary environment variables.
+    3. Setup database by running migrations
+       ```
+       pnpm drizzle:apply-migration
+       ```
+
+4. #### Run the Development Server
+
+```bash
+pnpm run dev
+```
+
+Visit [http://localhost:3000](http://localhost:3000/) in your browser to see the
+application.
+
+5. #### Build for Production
+
+```bash
+pnpm run build
+```
+
+## Resources for Beginners 📚
```

Comment:
I think one link for learning git is enough, You can add some of the articles from the `Readme.md` file.

https://github.com/growupanand/ConvoForm#learnings